File: backend/app/models/vehicle.py
"""車両モデル."""

from uuid import UUID

from sqlalchemy import Float, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, SoftDeleteMixin, TimestampMixin, UUIDPKMixin


class Vehicle(UUIDPKMixin, TimestampMixin, SoftDeleteMixin, Base):
    """車両モデル."""

    __tablename__ = "vehicle"

    user_id: Mapped[UUID] = mapped_column(index=True)
    name: Mapped[str] = mapped_column(String(255))
    seq: Mapped[int] = mapped_column(Integer)
    maker: Mapped[str] = mapped_column(String(100))
    model: Mapped[str] = mapped_column(String(100))
    year: Mapped[int | None] = mapped_column(Integer)
    number: Mapped[str | None] = mapped_column(String(50))
    tank_capacity: Mapped[float | None] = mapped_column(Float)

File: frontend/src/lib/types/vehicle.ts
import type { SuccessResponse } from './api'

export interface Vehicle {
  id: string
  user_id: string
  name: string
  seq: number
  maker: string
  model: string
  year: number | null
  number: string | null
  tank_capacity: number | null
  created_at: string
  updated_at: string
  deleted_at?: string | null
}

export interface CreateVehicleInput {
  name: string
  maker: string
  model: string
  year?: number | null
  number?: string | null
  tank_capacity?: number | null
}

export interface UpdateVehicleInput {
  name?: string | null
  seq?: number | null
  maker?: string | null
  model?: string | null
  year?: number | null
  number?: string | null
  tank_capacity?: number | null
}

export type VehicleResponse = SuccessResponse<Vehicle>
export type VehiclesResponse = SuccessResponse<Vehicle[]>
